.github/workflows/ci.yaml

```diff
@@ -4,142 +4,159 @@ on:
-        id: execution_timestamps
-        run: |
-          echo "::set-output name=starttime::$(curl -H "Accept: application/vnd.github+json" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq '.jobs[0].started_at')" && echo "::set-output name=endtime::$(curl -H "Accept: application/vnd.github+json" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq '.jobs[0].completed_at')"
-      
-      - name: calculate time
-        id: calculate_time
-        run: |
-          started_at=${{steps.execution_timestamps.outputs.starttime}}
-          completed_at=${{steps.execution_timestamps.outputs.endtime}}
+      - name: build
+        run: npm run build
+
+      #---Build Docker image, Login & Publish to remote registry---
+      - name: login #log in to the container registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login $REGISTRY -u ${{ github.actor }} --password-stdin
+
+      - name: build image
+        run: docker build -t $REGISTRY/$IMAGE_NAME:${GITHUB_SHA::7} -t $REGISTRY/$IMAGE_NAME:latest .
 
-          START_TIME=$(date -d "$started_at" +%s)
-          END_TIME=$(date -d "$completed_at" +%s)
+      - name: push image
+        run: docker push $REGISTRY/$IMAGE_NAME:${GITHUB_SHA::7} && docker push $REGISTRY/$IMAGE_NAME:latest
+
+  
+  stats:
+    timeout-minutes: 30
+    runs-on: ubuntu-20.04
+    if: ${{ always() }}
+    needs: [build, test]
+    
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
 
-          ELAPSE=$(( $END_TIME - $START_TIME ))
-          echo $ELAPSE
-          echo "$(($ELAPSE/60/60))h $(($ELAPSE/60%60))m $(($ELAPSE%60))s"
-          echo "::set-output name=elpsd::$ELAPSE"
-          echo "::set-output name=el_hours::$(($ELAPSE/60/60))"
-          echo "::set-output name=el_minutes::$(($ELAPSE/60%60))"
-          echo "::set-output name=el_seconds::$(($ELAPSE%60))"
+      - name: get email
+        id: email
+        run: |
+          if [[ ${{ github.event_name }} == 'pull_request' ]]; then
+            echo "::set-output name=user_email::$(git log -1 --pretty=format:%ae)" 
+          else
+            echo "::set-output name=user_email::${{ github.event.pusher.email }}"
+          fi
 
-      - name: Get job conclusion (success or failure)
+      - name: get job result
         id: job_conclusion
         run: |
-          echo "::set-output name=main_job_conclusion::$(curl -H "Accept: application/vnd.github+json" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq '.jobs[0].conclusion')"
+          for i in {0..1}
+          do
+          jobi=job_${i}_conclusion
+          echo "::set-output name=${jobi}::$(curl -H "Accept: application/vnd.github+json" \
+          https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs \
+          | jq --argjson i "$i" '.jobs[$i].conclusion')"
```

Comment:
1. Pls always offset text between do/done blocks, so it's easier to read,
2. this is not optimal approach for a few reasons. Your steps (and implementation) should not depends on the structure of the pipeline. I.e your step should not even know how many jobs/steps you got. The pipeline should work the same way if you refactor the number of jobs/steps, change their namings etc. (exception is only in `needs`).